File: app/src/main/java/info/nightscout/androidaps/plugins/pump/insight/database/InsightBolusID.java
package info.nightscout.androidaps.plugins.pump.insight.database;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import info.nightscout.androidaps.db.DatabaseHelper;

@DatabaseTable(tableName = DatabaseHelper.DATABASE_INSIGHT_BOLUS_IDS)
public class InsightBolusID {

    @DatabaseField(generatedId = true)
    public Long id;

    @DatabaseField
    public String pumpSerial;

    @DatabaseField
    public Long timestamp;

    @DatabaseField
    public Integer bolusID;

    @DatabaseField
    public Long startID;

    @DatabaseField
    public Long endID;

}

File: insight/src/main/java/info/nightscout/androidaps/insight/database/InsightBolusID.kt
package info.nightscout.androidaps.insight.database

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = DATABASE_INSIGHT_BOLUS_IDS,
    indices = [Index("bolusID")])
data class InsightBolusID(
    var timestamp: Long,
    var pumpSerial: String? = "None",
    var bolusID: Int? = null,
    var startID: Long? = null,
    var endID: Long? = null
) {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
}